BOJ/231204_숨바꼭질/kwaksj329.js

```diff
@@ -0,0 +1,23 @@
+const [subin, sis] = require("fs").readFileSync("/dev/stdin").toString().split(" ").map(Number);
+const visited = Array.from({ length: 100001 }, () => 0);
+
+function solution(subin, sis) {
+  const queue = [subin];
+  while (queue.length > 0) {
+    subin = queue.shift();
```

Comment:
`Array.prototype.shift()`는 python의 `popleft`와 다르게 `O(N)` 시간 복잡도를 가지기 때문에 데이터의 크기가 큰 경우에는 인덱스를 사용해 공간 복잡도를 늘리는 대신 시간 복잡도를 `O(1)`로 만들 수 있어요!
```suggestion
  while (index < queue.length) {
    subin = queue[index++];
```